### DISCUSSION TITLE
One screen to display & edit multiple products

### DISCUSSION THREAD

**Discussion by BloemenBezorgen** [Budibase/budibase#8362] [Help]
Hi there, I would like to set up budibase with a REST API connection to edit product price and product stock on my woocommerce shop. 

It would like to create a screen were I can list all the products in a table (name, price, on stock/off stock image) AND edit them as well. Like I can do in Excel. Would it be possible with budibase?

**Comment by melohagan** [comment]
Hey @BloemenBezorgen 

You can certainly create apps to display filtered data in a table, as well as screens to create new entries, and edit existing entries. 
With that said, there is currently no 'spreadsheet-like' functionality within Budibase where you can edit cells of data.

**Comment by BloemenBezorgen** [comment]
Hi Melohagan, thank you for your answer. I will than create two screens;

1 product list screen
1 product edit screen

I managed to create the product list screen with filters already. It would
be great to add a product image to the list. The Rest API is showing the
image url in "src":

"images": [
    {
      "id": 42806,
      "date_created": "2022-03-02T23:24:48",
      "date_created_gmt": "2022-03-02T21:24:48",
      "date_modified": "2022-03-02T23:24:48",
      "date_modified_gmt": "2022-03-02T21:24:48",
      "src": "
https://woocommerce-470197-2486223.cloudwaysapps.com/wp-content/uploads/2022/03/Rouwlint-groen-4-1-1.jpg
",
      "name": "rouwlint groen 4 1 1.jpg",
      "alt": "rouwlint groen 4 1 1.jpg"
    }
  ]

But I can't select the src field in the screens. How can I extract the
field from the REST API call?



Met vriendelijke groeten,

*Jan Pieters*


Op ma 24 okt. 2022 om 17:47 schreef melohagan ***@***.***>:

**Reply by melohagan** [reply]
Hey Jan. You could use the [transformer](https://docs.budibase.com/docs/transformers) to get the array of images to be your datasource.

E.g. in your REST query transformer:
```js
return data.images
```

Then in your screen you would have:
> Data Provider (rest query)
> > Repeater
> > > Image component (with the URL field bound to New Repeater.src)
